src/Tools/ErrorBoundary.js

```diff
@@ -0,0 +1,43 @@
+import propTypes from 'prop-types';
+import { Component } from 'react';
+
+import ErrorPage from '../PagesBody/ErrorPage/ErrorPage';
+
+export class ErrorBoundary extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { hasError: false, error: '', errorInfo: '' };
+  }
+
+  static getDerivedStateFromError(error) {
+    console.log(error);
+    return { hasError: true };
+  }
+
+  componentDidCatch(error, errorInfo) {
+    this.setState({
+      error: error,
+      errorInfo: errorInfo,
+    });
+    // console.log(error, errorInfo);
+  }
+
+  render() {
+    if (this.state.hasError) {
+      return (
+        <>
+          <ErrorPage />
+          <h4>Error: </h4> {this.state.error.toString()} <br />
+          <h4>Error Info: </h4> {this.state.errorInfo.componentStack}
+        </>
+      );
+    }
+    return this.props.children;
+  }
+}
+
+ErrorBoundary.propTypes = {
+  children: propTypes.node,
+};
+
+export default ErrorBoundary;
```

Comment:
Error Boundary works properly!
To make it better (it could be as TODO or PART2) - maybe it will be useful to add some redirection? to homepage or to previous page?
Error Boudnary is only used on test page? it is for some reason? maybe we can put inside error boundary our app?

Reply:
Error page will be used as a fallback from ErrorBoundary and there will be redirection. UI and Design is almost ready - Alicja is working on it. Its used on test page because our app is perfect and dont have any errors.

Reply:
I think its better to use ErrorBoudary on a components that might cause some errors. Not on the entire App. But this is case to consider.